src/TaskEither.ts

```diff
@@ -34,246 +29,141 @@ export const URI = 'TaskEither'
 
-const map = <L, A, B>(fa: TaskEither<L, A>, f: (a: A) => B): TaskEither<L, B> => {
-  return fa.map(f)
+/**
+ * @since 2.0.0
+ */
+export function foldTask<L, A, R>(
+  ma: TaskEither<L, A>,
+  onLeft: (l: L) => Task<R>,
+  onRight: (a: A) => Task<R>
+): Task<R> {
+  return task.chain(ma, e => E.fold(e, onLeft, onRight))
 }
 
 /**
  * @since 2.0.0
  */
-export const make = <A>(a: A): TaskEither<never, A> => {
-  return new TaskEither(T.of(a))
+export function getOrElse<L, A>(ma: TaskEither<L, A>, a: A): Task<A> {
+  return getOrElseL(ma, () => a)
 }
 
-const ap = <L, A, B>(fab: TaskEither<L, (a: A) => B>, fa: TaskEither<L, A>): TaskEither<L, B> => {
-  return fa.ap(fab)
+/**
+ * @since 2.0.0
+ */
+export function getOrElseL<L, A>(ma: TaskEither<L, A>, f: (l: L) => A): Task<A> {
+  return fold(ma, f, identity)
 }
 
-const chain = <L, A, B>(fa: TaskEither<L, A>, f: (a: A) => TaskEither<L, B>): TaskEither<L, B> => {
-  return fa.chain(f)
+/**
+ * @since 2.0.0
+ */
+export function mapLeft<L, A, M>(ma: TaskEither<L, A>, f: (l: L) => M): TaskEither<M, A> {
+  return task.map(ma, e => E.mapLeft(e, f))
 }
 
-const alt = <L, A>(fx: TaskEither<L, A>, fy: TaskEither<L, A>): TaskEither<L, A> => {
-  return fx.alt(fy)
+/**
+ * @since 2.0.0
+ */
+export function filterOrElse<L, A, B extends A>(ma: TaskEither<L, A>, p: Refinement<A, B>, zero: L): TaskEither<L, B>
+export function filterOrElse<L, A>(ma: TaskEither<L, A>, p: Predicate<A>, zero: L): TaskEither<L, A>
+export function filterOrElse<L, A>(ma: TaskEither<L, A>, p: Predicate<A>, zero: L): TaskEither<L, A> {
+  return task.map(ma, e => E.filterOrElse(e, p, zero))
 }
 
-const bimap = <L, V, A, B>(fa: TaskEither<L, A>, f: (l: L) => V, g: (a: A) => B): TaskEither<V, B> => {
-  return fa.bimap(f, g)
+/**
+ * @since 2.0.0
+ */
+export function filterOrElseL<L, A, B extends A>(
+  ma: TaskEither<L, A>,
+  p: Refinement<A, B>,
+  zero: (a: A) => L
+): TaskEither<L, B>
+export function filterOrElseL<L, A>(ma: TaskEither<L, A>, p: Predicate<A>, zero: (a: A) => L): TaskEither<L, A>
+export function filterOrElseL<L, A>(ma: TaskEither<L, A>, p: Predicate<A>, zero: (a: A) => L): TaskEither<L, A> {
+  return task.map(ma, e => E.filterOrElseL(e, p, zero))
 }
 
 /**
- * @since 1.0.0
+ * @since 2.0.0
  */
-export const right = <A>(fa: Task<A>): TaskEither<never, A> => {
-  return new TaskEither(fa.map(eitherRight))
-}
+export const fromRight: <A>(a: A) => TaskEither<never, A> = eitherT.of
 
 /**
- * @since 1.0.0
+ * @since 2.0.0
  */
-export const left = <L>(fl: Task<L>): TaskEither<L, never> => {
-  return new TaskEither(fl.map(eitherLeft))
+export function orElse<L, A, M>(ma: TaskEither<L, A>, f: (l: L) => TaskEither<M, A>): TaskEither<M, A> {
+  return task.chain(ma, e => E.fold<L, A, Task<E.Either<M, A>>>(e, f, eitherT.of))
 }
 
 /**
- * @since 1.0.0
+ * @since 2.0.0
  */
-export const fromEither = <L, A>(fa: Either<L, A>): TaskEither<L, A> => {
```

Comment:
this is now `taskEither.fromEither` from `MonadThrow`, correct?
It seems a bit weird to only have this removed (and not e.g. `fromLeft` which would be already available as `taskEither.throwError`)

Reply:
@giogonzo `fromLeft` and `taskEither.throwError` have different signatures (i.e. `never` in the former).

We can re-add a redundant `fromEither` later, I just want a clean baseline to start from.